crates/zk-por-cli/src/prover.rs

```diff
@@ -367,6 +369,39 @@ fn dump_proofs(
         .write_all(json!(root_proof).to_string().as_bytes())
         .map_err(|e| return PoRError::Io(e))?;
 
+    ///////////////////////////////////////////////
+    let hash_offset = RecursiveTargets::<RECURSION_BRANCHOUT_NUM>::pub_input_hash_offset();
+    let root_hash = HashOut::<F>::from_partial(&root_proof.proof.public_inputs[hash_offset]);
+    let root_hash_bytes = root_hash
+        .elements
+        .iter()
+        .map(|x| x.to_canonical_u64().to_le_bytes())
+        .flatten()
+        .collect::<Vec<u8>>();
+    let root_hash = hex::encode(root_hash_bytes);
+
+    let equity_offset = RecursiveTargets::<RECURSION_BRANCHOUT_NUM>::pub_input_equity_offset();
+    let equity_sum = root_proof.proof.public_inputs[equity_offset].to_canonical_u64();
+
+    let debt_offset = RecursiveTargets::<RECURSION_BRANCHOUT_NUM>::pub_input_debt_offset();
```

Comment:
offsets should be constants? It seems a bit weird logically to get offsets from targets.

Reply:
offsets is determined by targets. So it is better to put offset close to targets.